security/oc_tls.c

```diff
@@ -446,8 +432,10 @@ oc_tls_free_peer(oc_tls_peer_t *peer, bool inactivity_cb)
 
   size_t device = peer->endpoint.device;
   const oc_sec_pstat_t *pstat = oc_sec_get_pstat(device);
-  if (pstat->s == OC_DOS_RFOTM) {
-    oc_set_delayed_callback((void *)device, &reset_in_RFOTM, 0);
+  // Reset the device when the device onboarding connection
+  // has been closed and the device state is RFOTM.
+  if (peer->doc && pstat->s == OC_DOS_RFOTM && !reset) {
```

Comment:
This looks weird, the only use of the `reset` variable is to prevent device reset?

Reply:
reset is used when the tls connection is closed by calling `oc_pstat_handle_state - reset` from pstat.
It means to don't create something like:
`oc_pstat_handle_state - reset` -> `oc_close_all_tls_sessions_for_device_reset` -> `oc_tls_free_peer` -> oc_delayed_callback(`oc_pstat_handle_state - reset`, 2 *time.Second)